ospd_openvas/daemon.py

```diff
@@ -442,6 +442,116 @@ def scheduler(self):
+            _summary = _custom.pop('summary')
+        if 'impact' in _custom:
+            _impact = _custom.pop('impact')
+        if 'affected' in _custom:
+            _affected = _custom.pop('affected')
+        if 'insight' in _custom:
+            _insight = _custom.pop('insight')
+        if 'solution' in _custom:
+            _solution = _custom.pop('solution')
+            if 'solution_type' in _custom:
+                _solution_t = _custom.pop('solution_type')
+
+        if 'vuldetect' in _custom:
+            _vuldetect = _custom.pop('vuldetect')
+        if 'qod_type' in _custom:
+            _qod_t = _custom.pop('qod_type')
+        elif 'qod' in _custom:
+            _qod_v = _custom.pop('qod')
+
+        _severity = dict()
+        if 'severity_base_vector' in _custom:
+            _severity_vector = _custom.pop('severity_base_vector')
+        else:
+            _severity_vector = _custom.pop('cvss_base_vector')
+        _severity['severity_base_vector'] = _severity_vector
+        if 'severity_type' in _custom:
+            _severity_type = _custom.pop('severity_type')
+        else:
+            _severity_type = 'cvss_base_v2'
+        _severity['severity_type'] = _severity_type
+        if 'severity_origin' in _custom:
+            _severity['severity_origin'] = _custom.pop('severity_origin')
+
+        if _name is None:
+            name = ''
+
+        vt = {'name': _name}
+        if _custom is not None:
+            vt["custom"] = _custom
+        if _vt_params is not None:
+            vt["vt_params"] = _vt_params
+        if _vt_refs is not None:
+            vt["vt_refs"] = _vt_refs
+        if _vt_dependencies is not None:
+            vt["vt_dependencies"] = _vt_dependencies
+        if _vt_creation_time is not None:
+            vt["creation_time"] = _vt_creation_time
+        if _vt_modification_time is not None:
+            vt["modification_time"] = _vt_modification_time
+        if _summary is not None:
+            vt["summary"] = _summary
+        if _impact is not None:
+            vt["impact"] = _impact
+        if _affected is not None:
+            vt["affected"] = _affected
+        if _insight is not None:
+            vt["insight"] = _insight
+
+        if _solution is not None:
+            vt["solution"] = _solution
+            if _solution_t is not None:
+                vt["solution_type"] = _solution_t
+
+        if _vuldetect is not None:
+            vt["detection"] = _vuldetect
+
+        if _qod_t is not None:
+            vt["qod_type"] = _qod_t
+        elif _qod_v is not None:
+            vt["qod"] = _qod_v
+
+        if _severity is not None:
+            vt["severities"] = _severity
+
+        return vt
+
+    def get_vt_iterator(self):
+        """ Yield the vts from the Redis NVTicache. """
+        oids = dict(self.nvti.get_oids())
+        for filename, vt_id in oids.items():
+            vt = self.get_single_vt(vt_id, oids)
+            yield (vt_id, vt)
+
```

Comment:
For me this is not something the daemon class should care off. Personally I would move this functionality in the NVTICache class.

Reply:
Yes, sure. I am thinking on moving other methods to another files. But I would prefer to do the cleanup in a separated PR.

Reply:
Fine :-)